### DISCUSSION TITLE
Crear enlaces de descarga directa en OneDrive

### DISCUSSION THREAD

**Discussion by giscus** [zkreations/data#33] [Announcements]
# Crear enlaces de descarga directa en OneDrive

Alojar archivos en la nube y crear enlace de descarga directa para usarlos en el blog

https://www.zkreations.com/2024/02/crear-enlace-descarga-directa-onedrive.html

**Comment by aleh3101** [comment]
La otra opción usando el modo insertar y agregarlo al "/download?resid=" lo van a quitar? :0

**Reply by danieIabel** [reply]
Que tal amigo. Al menos a mi me dejo de funcionar ese método con los archivos que habia subido, por eso tuve que actualizar el tutorial.

**Reply by aleh3101** [reply]
Después de tiempo ya me empezó a salir los enlaces así. Pero se puede usar los datos de los nuevos enlaces en el antiguo link.  "onedrive.live.com/download?resid=xxxxx&authkey=xxxx"  sin el ASPX "download.aspx?authkey=xxxx&resId=xxx&cid=xxx"  Me pregunto si la parte de cid es necesario? porque solo tome el authkey y resid y me funciono la descarga 🤔

**Reply by danieIabel** [reply]
Si amigo tambien me di cuenta que funcionaba así, pero como no tenemos ni idea si algo puede llegar a cambiar, mejor mantenemos la mayor cantidad de datos posible en los enlaces, por ejemplo, supongamos que tiempo después los enlaces que no dispongan del atributo cid dejaran de funcionar? tendrias que cambiar todos los enlaces de tu blog. Es por eso que lo mantengo asi mas que nada para prevenir cualquier cosa a futuro, o al menos lo máximo posible.

**Comment by aleh3101** [comment]
Hola volvieron a cambiar los enlaces de descarga directa :(  Ahora me aparecen con y no se puede sacar para una descarga directa. https://1drv.ms/u/c/XXX/XXX

**Reply by danieIabel** [reply]
Que tal amigo, tecnicamente esta igual, solo debe visitar el enlace usando el modo incognito y esperar a que obtenga el enlace de respuesta.

**Reply by aleh3101** [reply]
No funciona porque me da los archivos con https://onedrive.live.com/embed?id=XXXX&resid=XXXX&cid=XXXX&redeem=XXXX&migratedtospo=true&embed=1, No me da los enlaces para descarga directa

**Reply by aleh3101** [reply]
Intente creando una cuenta nueva y ya me daban los enlaces como antes pero despues de unos dias volvia a pasar lo mismo dejaban de funcionar y cuando volvia a poner Insertar me daban los nuevos enlaces embed

**Reply by aleh3101** [reply]
Ejem: https://onedrive.live.com/download?resid=A8ED9A12D36BF959%21263&authkey=!ADJlD6BYk2zIecg
Entro a la cuenta y si esta el archivo y al darle Insertar me da este nuevo enlace:
https://1drv.ms/u/c/a8ed9a12d36bf959/UQNZ-WvTEprtIICoBwEAAAAAADJlD6BYk2zIecg y cuando lo abres da un enlace embed y no una descarga directa :s

**Reply by danieIabel** [reply]
Hola amigo, acabo de probar con otro enlace y sigue funcionando. ¿Que tipo de archivo subiste? quizas esa sea la razon de que no funcione.

**Reply by aleh3101** [reply]
Todos los archivos son canciones mp3. Cree una nueva cuenta ya que tenia problemas con una de Enero, entonces en esa nueva cuenta subia las canciones y ya me daba los enlaces con resid y authkey pero de la nada pasaban unos días, y cambiaban los enlaces de todos los archivos, asi como ves el error de que se debe poner en contacto, y al momento de darle insertar me daban el nuevo formato de url y al entrar en ese nuevo formato (embed?id=0000&resid=000&cid=000&redeem=000) no existe el authkey y por eso no me deja hacer la url de descarga directa.

**Reply by danieIabel** [reply]
Que tal amigo, he hecho exactamente lo que ha indicado (crear una nueva cuenta y subir un archivo mp3). Al pulsar sobre el archivo y pulsar sobre compartir se obtiene este enlace:

https://1drv.ms/u/s!AkPYzJf2PqBma83Ma15SSGJZovk?e=cVrrdW

Al visitar ése enlace en modo incognito se obtiene el siguiente enlace:

https://onedrive.live.com/?authkey=%21AM3Ma15SSGJZovk&id=66A03EF697CCD843%21107&cid=66A03EF697CCD843&parId=root&parQt=sharedby&o=OneUp

Como puede notar está todo lo necesario para que funcione el enlace de descarga directa. Quizás La confusion es que usted está intentando obtener los enlaces usando la opción "Insertar".

En el artículo estoy explicando el método nuevo ¿Usted se refiere al método antiguo? Tal vez por eso no le entendia, ya que ese método desde que actualice el articulo yo ya lo había descartado.

**Reply by aleh3101** [reply]
Usando compartir me da esto:      https://onedrive.live.com/?id=D820DC5033C8254D!311&resid=D820DC5033C8254D!311&cid=d820dc5033c8254d&redeem=aHR0cHM6Ly8xZHJ2Lm1zL3UvYy9kODIwZGM1MDMzYzgyNTRkL0VVMGx5RE5RM0NBZ2dOZzNBUUFBQUFBQkpzX3dfOVYxUkZSel9lc0pnUGFPaFE_ZT1lWHppcHg&migratedtospo=true   y si te das cuenta no existe el authkey, y este problema me ha estado pasando luego de varios días de haber creado una cuenta. Y algunas cuentas antiguas de 2 - 3 años le esta empezando a dar el mismo problema dejan de funcionar los enlaces ya que los archivos ya no cuentan con el authkey :s, y la unica solución que había tomado era crear nueva cuenta y volver a subir los audios pero ya cansa porque despues de unos dias volvía a pasar eso en esas cuentas nuevas.

**Reply by danieIabel** [reply]
Buenas amigo, ¿me podrias dar acceso a esa cuenta nueva por privado bajo tu consentimiento? para así poder investigar el asunto. Puedes ponerte en contacto conmigo por telegram o el formulario de contacto. De todas formas esperare unos días para intentar de nuevo con mi cuenta nueva.
